pco/src/float_quant_utils.rs

```diff
@@ -104,6 +104,28 @@ mod test {
 
   use super::*;
 
+  #[test]
+  fn test_estimate_best_k() {
+    // all but the last of these have 21 out of 23 mantissa bits zeroed
+    let mut sample = vec![1.0_f32, 1.25, -1.5, 1.75, -0.875, 0.75, 0.625].repeat(3);
+    sample.push(f32::from_bits(1.0_f32.to_bits() + 1));
+    let (k, freq) = estimate_best_k_and_freq(&sample);
+    assert_eq!(k, 21);
+    assert_eq!(
+      freq,
+      (sample.len() - 1) as f64 / (sample.len() as f64)
+    );
```

Comment:
I'd suggest explicitly showing the dependence on `REQUIRED_QUANTIZED_PROPORTION` so that readers don't miss it:
```suggestion
    );
    assert!(freq >= REQUIRED_QUANTIZED_PROPORTION);
```